agent/app/agent_unix_test.go

```diff
@@ -115,13 +114,20 @@ func TestDoStartHappyPath(t *testing.T) {
 		mobyPlugins:        mockMobyPlugins,
 	}
 
-	go agent.doStart(eventstream.NewEventStream("events", ctx),
-		credentialsManager, state, imageManager, client)
+	var agentW sync.WaitGroup
+	agentW.Add(1)
+	go func() {
+		agent.doStart(eventstream.NewEventStream("events", ctx),
```

Comment:
there are some more tests running agent.doStart in agent_test.go and agent_windows_test.go. it's better to fix them together?

Reply:
👍 , I can fix those too

Reply:
actually just one, TestDoStartRegisterAvailabilityZone

Reply:
There was only one other instance in agent_test.go and it's fixed now too, the other  instances wait to check the status code so they are OK.